sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/AesCbcHmacSha2.java

```diff
@@ -0,0 +1,270 @@
+            throw new IllegalArgumentException("No initialization vector");
+        }
+
+        if (authenticationData == null) {
+            throw new IllegalArgumentException("No authentication data");
+        }
+
+        if (authenticationTag == null) {
+            throw new IllegalArgumentException("No authentication tag");
+        }
+
+        // Create the Decryptor
+        return new AesCbcHmacSha2Decryptor(getName(), key, iv, authenticationData, authenticationTag, provider);
+    }
+
+    @Override
+    public ICryptoTransform CreateEncryptor(byte[] key, byte[] iv, byte[] authenticationData) throws InvalidKeyException, NoSuchAlgorithmException, NoSuchPaddingException, InvalidAlgorithmParameterException {
+        return CreateEncryptor(key, iv, authenticationData, null);
+    }
+
+    @Override
+    public ICryptoTransform CreateEncryptor(byte[] key, byte[] iv, byte[] authenticationData, Provider provider) throws InvalidKeyException, NoSuchAlgorithmException, NoSuchPaddingException, InvalidAlgorithmParameterException {
+
+        if (key == null) {
+            throw new IllegalArgumentException("No key material");
+        }
+
+        if (iv == null) {
+            throw new IllegalArgumentException("No initialization vector");
+        }
+
+        if (authenticationData == null) {
+            throw new IllegalArgumentException("No authentication data");
+        }
+
+        // Create the Encryptor
+        return new AesCbcHmacSha2Encryptor(getName(), key, iv, authenticationData, provider);
+    }
+
+    private static Triple<byte[], byte[], Mac> getAlgorithmParameters(String algorithm, byte[] key) throws InvalidKeyException, NoSuchAlgorithmException {
+
+        byte[] aesKey;
+        byte[] hmacKey;
+        Mac hmac;
+
+        if (algorithm.equalsIgnoreCase(Aes128CbcHmacSha256.ALGORITHM_NAME)) {
+            if ((key.length << 3) < 256) {
+                throw new IllegalArgumentException(String.format("%s key length in bits %d < 256", algorithm, key.length << 3));
+            }
+
+            hmacKey = new byte[128 >> 3];
+            aesKey = new byte[128 >> 3];
+
+            // The HMAC key precedes the AES key
+            System.arraycopy(key, 0, hmacKey, 0, 128 >> 3);
+            System.arraycopy(key, 128 >> 3, aesKey, 0, 128 >> 3);
+
+            hmac = Mac.getInstance("HmacSHA256");
+            hmac.init(new SecretKeySpec(hmacKey, "HmacSHA256"));
+
+        } else if (algorithm.equalsIgnoreCase(Aes192CbcHmacSha384.ALGORITHM_NAME)) {
+
+            if ((key.length << 3) < 384) {
+                throw new IllegalArgumentException(String.format("%s key length in bits %d < 384", algorithm, key.length << 3));
+            }
+
+            hmacKey = new byte[192 >> 3];
+            aesKey = new byte[192 >> 3];
+
+            // The HMAC key precedes the AES key
+            System.arraycopy(key, 0, hmacKey, 0, 192 >> 3);
+            System.arraycopy(key, 192 >> 3, aesKey, 0, 192 >> 3);
```

Comment:
This looks like 192 bits size symmetric key.
Currently it makes it easy to read that 192 bits size is getting converted to bytes via bitwise op. with constants it will be confusing to read because of bitwise expressions.  
Plus its internal.